worker/changestream/worker.go

```diff
@@ -131,13 +122,19 @@ func (w *changeStreamWorker) Report() map[string]any {
 
 // GetWatchableDB returns a new WatchableDB for the given namespace.
 func (w *changeStreamWorker) GetWatchableDB(namespace string) (changestream.WatchableDB, error) {
-	db, err := w.cfg.DBGetter.GetDB(namespace)
-	if err != nil {
-		return nil, errors.Trace(err)
+	// If the worker already exists, return the existing worker early.
+	if mux, err := w.runner.Worker(namespace, w.catacomb.Dying()); err == nil {
```

Comment:
You specifically check for `ErrDead` in the db-accessor worker. Do you need to do that here?

I think both here and in that worker we should be proceeding on to `StartWorker` only if we got a `NotFound` error.

Reply:
I've just normalized the two, so the behave in the same way.

Reply:
We could make a generic function from this, but it seems a bit overkill imo.